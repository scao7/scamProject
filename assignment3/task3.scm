;constructor
(define (node inValue inParent inChild)
    (define val inValue)
    (define parent inParent)
    (define child inChild)
    (define (set-parent p) (set 'parent p))
    (define (set-child c) (set 'child c))
    (define (set-val va) (set 'val va))
    (define (get-parent) parent)
    (define (get-child) child)
    (define (get-val) val)
    this
)


(define (deque)
    (define queuelist nil)
    (define size 0)
    (define tail nil)
    (define (enqueueFront val)
        (define n (node val nil nil))
        (if (null? queuelist)
            (begin
                (set 'size 1)
                (set 'queuelist n)
                (set 'tail n)
                )
            (begin
                ((n 'set-child) queuelist)
                ((queuelist 'set-parent) n)
                (set 'queuelist n)
		(set! size (+ size 1))
                )
            )
        val
        )


    (define (enqueueBack val)
        (define n (node val nil nil))
        (if (null? queuelist)
            (begin
                (set 'size 1)
                (set 'queuelist n)
                (set 'tail n)
                )
            (begin
                ((n 'set-parent) tail)
                ((tail 'set-child) n)
                (set 'tail n)
		(set! size (+ size 1))
                )
            )
        val
        )


    (define (enqueueIndex ind val)
        (define (enqueIter src count in va)
            (if (= count in)
                (begin
                    (define c ((src 'get-child)))
                    (define n (node va src c))
                    ((src 'set-child) n)
                    (if (= in size)
                        (set 'tail n)
                        ((c 'set-parent) n)
                        )
                    )
                (enqueIter ((src 'get-child)) (+ count 1) in va)
                )
            )

        (if (= ind 0)
            (enqueueFront val)
            (begin
                (enqueIter queuelist 1 ind val)
		(set! size (+ size 1))
                )
            )
        val
        )


    (define (dequeueFront)
        (define v ((queuelist 'get-val)))
        (set 'queuelist ((queuelist 'get-child)))
	(set! size (- size 1))
        (if (= size 0)
            (set 'tail nil)
            ((queuelist 'set-parent) nil)
            )
        v
        )


    (define (dequeueBack)
        (define v ((tail 'get-val)))
        (set 'tail ((tail 'get-parent)))
	(set! size (- size 1))
        (if (= size 0)
            (set 'queuelist nil)
            ((tail 'set-child) nil)
            )
        v
        )


    (define (dequeueIndex ind)
        (define (displayIter src ind count)
            (if (= ind count)
                (begin
                    (define c ((src 'get-child)))
                    (define p ((src 'get-parent)))
                    (define v ((src 'get-val)))
                    ((c 'set-parent) p)
                    (if (!= count 0)
                        ((p 'set-child) c)
                        (set 'queuelist c)
                        )
                    v
                    )
                (displayIter ((src 'get-parent)) ind (- count 1))
                )
            )
        

        (if (= ind (- size 1))
            (dequeueBack)
            (begin
                (define v (displayIter ((tail 'get-parent)) ind (- size 2)))
		(set! size (- size 1))
                v
                )
            )
        )

    
    (define (display)
        (define (displayIter src)
            (if (not (null? src))
                (begin 
                    (print ((src 'get-val)))
                    (if (not (null? ((src 'get-child))))
                        (begin 
                            (print ",")
                            (displayIter ((src 'get-child)))
                            )
                        )
                    )
                )
            )

        (print "[")
        (displayIter queuelist)
        (print "]")
        )


    (define (peekFront)
        ((queuelist 'get-val))
        )

    
    (define (peekBack)
        ((tail 'get-val))
        )
        

    this
    )


(define (main)
    (define env this)
    (define (iter expr)
         (if (not (eof?)) (begin (eval expr env) (iter (readExpr))))
         )
    (setPort (open (getElement ScamArgs 1) 'read))
    (iter (readExpr))
    )
