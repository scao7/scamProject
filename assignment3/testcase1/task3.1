(define dq (deque))
((dq'enqueueIndex) 0 48)
((dq'enqueueIndex) 0 64)
((dq'enqueueIndex) 1 32)
((dq'enqueueIndex) 1 20)
((dq'enqueueIndex) 4 26)
((dq'enqueueIndex) 1 27)
((dq'enqueueIndex) 1 19)
((dq'enqueueIndex) 2 23)
((dq'enqueueIndex) 5 77)
((dq'enqueueIndex) 1 23)
((dq'display))
(println)
