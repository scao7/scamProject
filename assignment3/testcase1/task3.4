(define dq (deque))
((dq'enqueueIndex) 0 63)
((dq'enqueueIndex) 0 68)
((dq'enqueueIndex) 2 83)
((dq'enqueueIndex) 3 21)
((dq'enqueueIndex) 3 60)
((dq'enqueueIndex) 3 78)
((dq'enqueueIndex) 1 57)
((dq'enqueueIndex) 7 60)
((dq'enqueueIndex) 6 98)
((dq'enqueueIndex) 5 64)
((dq'enqueueIndex) 9 27)
((dq'enqueueIndex) 6 39)
((dq'enqueueIndex) 12 60)
((dq'enqueueIndex) 0 69)
((dq'enqueueIndex) 7 36)
((dq'enqueueIndex) 10 82)
((dq'enqueueIndex) 15 89)
((dq'enqueueIndex) 8 79)
((dq'enqueueIndex) 13 48)
((dq'enqueueIndex) 17 49)
((dq'enqueueIndex) 20 50)
((dq'enqueueIndex) 4 22)
((dq'enqueueIndex) 22 22)
((dq'enqueueIndex) 16 68)
((dq'enqueueIndex) 3 28)
((dq'enqueueIndex) 6 98)
((dq'enqueueIndex) 10 85)
((dq'enqueueIndex) 8 42)
((dq'enqueueIndex) 11 39)
((dq'enqueueIndex) 0 90)
((dq'enqueueIndex) 19 72)
((dq'enqueueIndex) 22 57)
((dq'enqueueIndex) 19 24)
((dq'enqueueIndex) 10 35)
((dq'enqueueIndex) 20 27)
((dq'enqueueIndex) 26 61)
((dq'enqueueIndex) 23 93)
((dq'enqueueIndex) 26 65)
((dq'enqueueIndex) 2 46)
((dq'enqueueIndex) 11 27)
((dq'enqueueIndex) 25 58)
((dq'enqueueIndex) 7 11)
((dq'enqueueIndex) 15 55)
((dq'enqueueIndex) 16 79)
((dq'enqueueIndex) 37 45)
((dq'enqueueIndex) 30 51)
((dq'enqueueIndex) 4 43)
((dq'enqueueIndex) 47 70)
((dq'enqueueIndex) 26 36)
((dq'enqueueIndex) 48 21)
((dq'enqueueIndex) 24 71)
((dq'enqueueIndex) 35 21)
((dq'enqueueIndex) 32 45)
((dq'enqueueIndex) 40 70)
((dq'enqueueIndex) 44 12)
((dq'enqueueIndex) 48 47)
((dq'enqueueIndex) 32 14)
((dq'enqueueIndex) 25 94)
((dq'enqueueIndex) 32 83)
((dq'enqueueIndex) 42 44)
((dq'enqueueIndex) 13 43)
((dq'enqueueIndex) 52 37)
((dq'enqueueIndex) 47 84)
((dq'enqueueIndex) 62 35)
((dq'enqueueIndex) 7 94)
((dq'enqueueIndex) 27 64)
((dq'enqueueIndex) 41 19)
((dq'enqueueIndex) 50 31)
((dq'enqueueIndex) 34 54)
((dq'enqueueIndex) 63 37)
((dq'enqueueIndex) 37 80)
((dq'enqueueIndex) 51 17)
((dq'enqueueIndex) 60 23)
((dq'enqueueIndex) 1 43)
((dq'enqueueIndex) 72 75)
((dq'enqueueIndex) 57 26)
((dq'enqueueIndex) 8 63)
((dq'enqueueIndex) 38 87)
((dq'enqueueIndex) 33 52)
((dq'enqueueIndex) 11 59)
((dq'enqueueIndex) 33 59)
((dq'enqueueIndex) 12 13)
((dq'enqueueIndex) 54 90)
((dq'enqueueIndex) 22 24)
((dq'enqueueIndex) 32 26)
((dq'enqueueIndex) 39 91)
((dq'enqueueIndex) 83 26)
((dq'enqueueIndex) 87 80)
((dq'enqueueIndex) 29 11)
((dq'enqueueIndex) 14 37)
((dq'display))
(println)
((dq'dequeueIndex) 67)
((dq'display))
(println)
((dq'dequeueIndex) 81)
((dq'display))
(println)
((dq'dequeueIndex) 42)
((dq'display))
(println)
((dq'dequeueIndex) 74)
((dq'display))
(println)
((dq'dequeueIndex) 44)
((dq'display))
(println)
((dq'dequeueIndex) 83)
((dq'display))
(println)
((dq'dequeueIndex) 59)
((dq'display))
(println)
((dq'dequeueIndex) 78)
((dq'display))
(println)
((dq'dequeueIndex) 36)
((dq'display))
(println)
((dq'dequeueIndex) 69)
((dq'display))
(println)
((dq'dequeueIndex) 40)
((dq'display))
(println)
((dq'dequeueIndex) 68)
((dq'display))
(println)
((dq'dequeueIndex) 31)
((dq'display))
(println)
((dq'dequeueIndex) 50)
((dq'display))
(println)
((dq'dequeueIndex) 68)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 40)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 15)
((dq'display))
(println)
((dq'dequeueIndex) 66)
((dq'display))
(println)
((dq'dequeueIndex) 24)
((dq'display))
(println)
((dq'dequeueIndex) 46)
((dq'display))
(println)
((dq'dequeueIndex) 57)
((dq'display))
(println)
((dq'dequeueIndex) 20)
((dq'display))
(println)
((dq'dequeueIndex) 56)
((dq'display))
(println)
((dq'dequeueIndex) 54)
((dq'display))
(println)
((dq'dequeueIndex) 6)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 53)
((dq'display))
(println)
((dq'dequeueIndex) 15)
((dq'display))
(println)
((dq'dequeueIndex) 29)
((dq'display))
(println)
((dq'dequeueIndex) 35)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 56)
((dq'display))
(println)
((dq'dequeueIndex) 25)
((dq'display))
(println)
((dq'dequeueIndex) 38)
((dq'display))
(println)
((dq'dequeueIndex) 51)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 33)
((dq'display))
(println)
((dq'dequeueIndex) 20)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 20)
((dq'display))
(println)
((dq'dequeueIndex) 37)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 21)
((dq'display))
(println)
((dq'dequeueIndex) 15)
((dq'display))
(println)
((dq'dequeueIndex) 14)
((dq'display))
(println)
((dq'dequeueIndex) 29)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 26)
((dq'display))
(println)
((dq'dequeueIndex) 15)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 35)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 33)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 14)
((dq'display))
(println)
((dq'dequeueIndex) 26)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 27)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 14)
((dq'display))
(println)
((dq'dequeueIndex) 24)
((dq'display))
(println)
((dq'dequeueIndex) 22)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 20)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 18)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 9)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 6)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'display))
(println)
((dq'enqueueIndex) 0 27)
((dq'enqueueIndex) 1 96)
((dq'enqueueIndex) 2 73)
((dq'enqueueIndex) 3 96)
((dq'enqueueIndex) 2 95)
((dq'enqueueIndex) 5 89)
((dq'enqueueIndex) 0 88)
((dq'enqueueIndex) 3 61)
((dq'enqueueIndex) 6 26)
((dq'enqueueIndex) 5 92)
((dq'enqueueIndex) 2 39)
((dq'enqueueIndex) 5 11)
((dq'enqueueIndex) 5 16)
((dq'enqueueIndex) 10 85)
((dq'enqueueIndex) 10 66)
((dq'enqueueIndex) 10 30)
((dq'enqueueIndex) 14 50)
((dq'enqueueIndex) 3 59)
((dq'enqueueIndex) 2 15)
((dq'enqueueIndex) 10 76)
((dq'enqueueIndex) 0 42)
((dq'enqueueIndex) 13 14)
((dq'enqueueIndex) 5 17)
((dq'enqueueIndex) 14 95)
((dq'enqueueIndex) 6 29)
((dq'enqueueIndex) 22 57)
((dq'enqueueIndex) 14 35)
((dq'enqueueIndex) 15 99)
((dq'enqueueIndex) 10 39)
((dq'enqueueIndex) 24 17)
((dq'enqueueIndex) 29 55)
((dq'enqueueIndex) 10 80)
((dq'enqueueIndex) 31 55)
((dq'enqueueIndex) 11 18)
((dq'enqueueIndex) 19 86)
((dq'enqueueIndex) 30 61)
((dq'enqueueIndex) 7 51)
((dq'enqueueIndex) 23 49)
((dq'enqueueIndex) 21 32)
((dq'enqueueIndex) 15 83)
((dq'enqueueIndex) 19 32)
((dq'enqueueIndex) 14 10)
((dq'enqueueIndex) 22 91)
((dq'enqueueIndex) 43 90)
((dq'enqueueIndex) 10 84)
((dq'enqueueIndex) 45 27)
((dq'enqueueIndex) 15 36)
((dq'enqueueIndex) 47 35)
((dq'enqueueIndex) 39 38)
((dq'enqueueIndex) 18 42)
((dq'enqueueIndex) 8 29)
((dq'enqueueIndex) 48 43)
((dq'enqueueIndex) 35 60)
((dq'enqueueIndex) 44 30)
((dq'enqueueIndex) 44 27)
((dq'enqueueIndex) 2 35)
((dq'enqueueIndex) 25 46)
((dq'enqueueIndex) 16 98)
((dq'enqueueIndex) 18 35)
((dq'enqueueIndex) 52 59)
((dq'enqueueIndex) 7 87)
((dq'enqueueIndex) 46 50)
((dq'enqueueIndex) 9 75)
((dq'enqueueIndex) 46 96)
((dq'enqueueIndex) 2 19)
((dq'enqueueIndex) 21 29)
((dq'enqueueIndex) 21 33)
((dq'enqueueIndex) 40 99)
((dq'enqueueIndex) 56 46)
((dq'enqueueIndex) 15 67)
((dq'enqueueIndex) 42 34)
((dq'enqueueIndex) 66 14)
((dq'enqueueIndex) 48 29)
((dq'enqueueIndex) 2 98)
((dq'enqueueIndex) 37 91)
((dq'enqueueIndex) 40 65)
((dq'enqueueIndex) 59 35)
((dq'enqueueIndex) 5 92)
((dq'enqueueIndex) 0 82)
((dq'enqueueIndex) 70 14)
((dq'enqueueIndex) 73 28)
((dq'enqueueIndex) 21 29)
((dq'enqueueIndex) 38 86)
((dq'enqueueIndex) 18 35)
((dq'enqueueIndex) 21 49)
((dq'enqueueIndex) 79 87)
((dq'enqueueIndex) 61 86)
((dq'enqueueIndex) 80 43)
((dq'enqueueIndex) 6 95)
((dq'enqueueIndex) 31 61)
((dq'display))
(println)
((dq'dequeueIndex) 77)
((dq'display))
(println)
((dq'dequeueIndex) 78)
((dq'display))
(println)
((dq'dequeueIndex) 16)
((dq'display))
(println)
((dq'dequeueIndex) 54)
((dq'display))
(println)
((dq'dequeueIndex) 13)
((dq'display))
(println)
((dq'dequeueIndex) 22)
((dq'display))
(println)
((dq'dequeueIndex) 45)
((dq'display))
(println)
((dq'dequeueIndex) 13)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 34)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 76)
((dq'display))
(println)
((dq'dequeueIndex) 48)
((dq'display))
(println)
((dq'dequeueIndex) 37)
((dq'display))
(println)
((dq'dequeueIndex) 14)
((dq'display))
(println)
((dq'dequeueIndex) 6)
((dq'display))
(println)
((dq'dequeueIndex) 25)
((dq'display))
(println)
((dq'dequeueIndex) 29)
((dq'display))
(println)
((dq'dequeueIndex) 26)
((dq'display))
(println)
((dq'dequeueIndex) 42)
((dq'display))
(println)
((dq'dequeueIndex) 58)
((dq'display))
(println)
((dq'dequeueIndex) 20)
((dq'display))
(println)
((dq'dequeueIndex) 31)
((dq'display))
(println)
((dq'dequeueIndex) 36)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 24)
((dq'display))
(println)
((dq'dequeueIndex) 58)
((dq'display))
(println)
((dq'dequeueIndex) 13)
((dq'display))
(println)
((dq'dequeueIndex) 19)
((dq'display))
(println)
((dq'dequeueIndex) 16)
((dq'display))
(println)
((dq'dequeueIndex) 47)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 9)
((dq'display))
(println)
((dq'dequeueIndex) 56)
((dq'display))
(println)
((dq'dequeueIndex) 44)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 13)
((dq'display))
(println)
((dq'dequeueIndex) 18)
((dq'display))
(println)
((dq'dequeueIndex) 25)
((dq'display))
(println)
((dq'dequeueIndex) 15)
((dq'display))
(println)
((dq'dequeueIndex) 38)
((dq'display))
(println)
((dq'dequeueIndex) 34)
((dq'display))
(println)
((dq'dequeueIndex) 13)
((dq'display))
(println)
((dq'dequeueIndex) 18)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 21)
((dq'display))
(println)
((dq'dequeueIndex) 20)
((dq'display))
(println)
((dq'dequeueIndex) 22)
((dq'display))
(println)
((dq'dequeueIndex) 36)
((dq'display))
(println)
((dq'dequeueIndex) 34)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 27)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 21)
((dq'display))
(println)
((dq'dequeueIndex) 9)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 32)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 16)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 13)
((dq'display))
(println)
((dq'dequeueIndex) 9)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 16)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 6)
((dq'display))
(println)
((dq'dequeueIndex) 22)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 13)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
