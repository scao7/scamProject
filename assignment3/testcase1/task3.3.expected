[44,40,24,67,26,44,11,12,10,48]
[44,40,24,26,44,11,12,10,48]
[44,40,24,26,44,11,12,48]
[44,24,26,44,11,12,48]
[44,24,26,44,11,12]
[44,24,26,11,12]
[44,24,26,12]
[44,26,12]
[44,12]
[44]
[]
[]
[69,43,10,27,45,51,85,27,83,93]
[69,43,10,27,45,85,27,83,93]
[69,43,10,27,45,85,27,93]
[69,10,27,45,85,27,93]
[69,10,27,45,85,93]
[69,10,27,45,93]
[69,10,27,93]
[69,27,93]
[27,93]
[93]
[]
